invenio_rdm_records/requests/user_moderation/actions.py

```diff
@@ -8,12 +8,48 @@
 
 from invenio_access.permissions import system_identity
 
-from invenio_rdm_records.proxies import current_rdm_records_service
+from ...proxies import current_rdm_records_service
 
 
 def on_block(user_id, uow=None, **kwargs):
     """Removes records that belong to a user."""
-    pass
+    user_id = str(user_id)
+    record_cls = current_rdm_records_service.record_cls
+    model_cls = record_cls.model_cls
+    parent_cls = record_cls.parent_record_cls
+    parent_model_cls = parent_cls.model_cls
+    # TODO check if the removal_reason vocabulary entry exists
+    # TODO maybe set a different removal_reason? take it from the kwargs?
+    tombstone_data = {
+        "note": "User was blocked",
+        "removal_reason": {"id": "misconduct"},
+    }
+
+    # get all the parent records owned by the blocked user
+    parent_recs = [
```

Comment:
these can be potentially costly queries to run, is this action done in celery task ?

Reply:
Yes, in the user's service, we kick off a celery task and execute all the actions inside it.

We can leave a comment here (e.g. `this is meant to be used ...`).

Reply:
Updated the docstring to reflect this!